packages/web-forms/src/components/common/map/vertex-geometry.ts

```diff
@@ -0,0 +1,228 @@
+import type { Coordinate } from 'ol/coordinate';
+import Feature from 'ol/Feature';
+import { Point, LineString, Polygon } from 'ol/geom';
+
+export const getFlatCoordinates = (geometry: LineString | Polygon | undefined) => {
+	if (!geometry) {
+		return [];
+	}
+
+	if (geometry instanceof LineString) {
+		return geometry.getCoordinates();
+	}
+
+	return geometry.getCoordinates()[0] ?? [];
+};
+
+const getDeltaBetweenPoints = (pointA: number[], pointB: number[]) => {
+	const isDefined = (c: number) => c != null;
+	const pairA = pointA.filter(isDefined) as [number, number];
+	const pairB = pointB.filter(isDefined) as [number, number];
+	if (pairA.length < 2 || pairB.length < 2) {
+		return { deltaX: 0, deltaY: 0 };
+	}
+
+	const [ax, ay] = pairA;
+	const [bx, by] = pairB;
+	return { deltaX: ax - bx, deltaY: ay - by };
+};
+
+const getDistanceBetweenPoints = (deltaX: number, deltaY: number) => {
```

Comment:
It took me a minute to work out why you were squaring the deltas before realising it's the Pythagorean theorem - could you add a comment?

NB: Very clever, but not clever in the show off kind of way, clever in the right way! Love it.